core/src/main/java/io/questdb/cairo/wal/WalUtils.java

```diff
@@ -96,4 +106,73 @@ public static void createTxnLogFile(FilesFacade ff, MemoryMARW mem, Path txnSeqD
             txnSeqDirPath.trimTo(rootLen);
         }
     }
+
+    /**
+     * Retrieves the last refresh base transaction for a materialized view.
+     *
+     * @param tablePath          the path to the table
+     * @param tableToken         the table token
+     * @param configuration      the Cairo configuration
+     * @param txnLogMemory       the memory to iterate over transaction logs
+     * @param walEventReader     the reader to read WAL events
+     * @param blockFileReader    the reader to read state file
+     * @param matViewStateReader the POD to read materialized view state into
+     * @return -1 if the transaction could not be extracted, -2 if the last WAL-E entry is an invalidation commit (MAT_VIEW_INVALIDATE),
+     * or a transaction number greater than -1 if it is a valid last refresh transaction
+     */
+    public static long getMatViewLastRefreshBaseTxn(
+            Path tablePath,
+            TableToken tableToken,
+            CairoConfiguration configuration,
+            MemoryCMR txnLogMemory,
+            WalEventReader walEventReader,
+            BlockFileReader blockFileReader,
+            MatViewStateReader matViewStateReader
+    ) {
+        long txnNotFound = -1;
```

Comment:
nit: could be a const, i.e. `static final` field, just like `txnInvalid`

Reply:
fixed: https://github.com/questdb/questdb/pull/5519/commits/80520bf2e0c98719697c493f944de2f989da2863